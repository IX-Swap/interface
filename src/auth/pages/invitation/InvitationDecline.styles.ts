import makeStyles from '@mui/styles/makeStyles'

export const useStyles = makeStyles(theme => ({
  wrapper: {
    display: 'block',
    textAlign: 'center'
  },
  title: {
    fontWeight: 700,
    fontSize: 40,
    color: theme.palette.text.secondary,
    textTransform: 'uppercase'
  },
  description: {
    display: 'block',
    margin: theme.spacing(0, 'auto'),
    color: theme.palette.text.primary
  },
  text: {
    display: 'block',
    margin: theme.spacing(0, 'auto'),
    color: theme.palette.text.primary
  },
  link: {
    color: theme.palette.text.secondary
  },
  customButton: {
    backgroundColor: '#0C469C',
    color: '#FFFFFF',
    width: '8rem',
    height: '3rem',
    padding: '0.5rem',
    fontSize: '0.8rem',
    lineHeight: '1rem',
    fontWeight: 700
  }
}))
