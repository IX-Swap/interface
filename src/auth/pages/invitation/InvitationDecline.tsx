import React, { Fragment, useEffect, useState } from 'react'
import { Divider, Grid, Typography, Button } from '@mui/material'
import { VSpacer } from 'components/VSpacer'
import { useStyles } from './InvitationDecline.styles'
import InvestaLogo from 'assets/icons/logo-color.svg'
import { useInvitation } from 'auth/hooks/useInvitation'
import { AppRouterLink } from 'components/AppRouterLink'
import { AuthRoute } from 'auth/router/config'
import { useParams } from 'react-router-dom'

export const InvitationDeclie = () => {
  const { roleId } = useParams<{
    roleId: string
  }>()
  const { wrapper, description, customButton, text } = useStyles()
  const [decline, { isSuccess, isError }] = useInvitation(roleId)
  const [state, setState] = useState({
    message: 'Are you sure you want to decline the invitation?',
    btnTxt: 'Yes, Decline'
  })
  useEffect(() => {
    isError &&
      setState({
        ...state,
        message: 'Something went wrong. Could not decline the invitation',
        btnTxt: 'Try Again'
      })
  }, [isError]) // eslint-disable-line

  const handleSubmit = async () => {
    await decline()
  }

  return (
    <Grid container className={wrapper}>
      <Grid item>
        <img alt='logo' src={InvestaLogo} />
        <VSpacer size={'medium'} />
      </Grid>
      {!isSuccess && (
        <Fragment>
          <Grid item>
            <Typography
              className={description}
              variant={'body1'}
              align='center'
            >
              {state.message}
            </Typography>
            <VSpacer size={'medium'} />
          </Grid>
          <Grid item>
            <Divider />
            <VSpacer size={'small'} />
          </Grid>
          <Grid item>
            <Button
              className={customButton}
              size='large'
              onClick={handleSubmit}
            >
              {state.btnTxt}
            </Button>
          </Grid>
        </Fragment>
      )}
      {isSuccess && (
        <Fragment>
          <Grid item>
            <Typography align='center' className={text}>
              Donâ€™t have an account?{' '}
              <AppRouterLink to={AuthRoute.signup}>
                Create an Account.
              </AppRouterLink>
            </Typography>
          </Grid>
          <br />
          <Grid item>
            <Typography align='center' className={text}>
              Already have an account?{' '}
              <AppRouterLink to={AuthRoute.login}>Sign In.</AppRouterLink>
            </Typography>
          </Grid>
        </Fragment>
      )}
    </Grid>
  )
}
