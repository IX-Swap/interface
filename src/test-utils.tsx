import React from 'react'
import { Router } from 'react-router-dom'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import {
  createGenerateClassName,
  StylesProvider
} from '@material-ui/core/styles'
import { ThemeProvider } from '@material-ui/styles'
import Themes from './v2/themes'
import { history } from 'v2/history'
import { UserProvider } from 'v2/auth/context'
import { UserStore } from 'v2/auth/context/store'
import { PasswordResetProvider } from 'v2/auth/context/password-reset'
import PasswordResetStore from 'v2/auth/context/password-reset/store'
import { PasswordResetStep } from 'v2/auth/context/password-reset/types'
import { AuthorizerTableStore } from 'v2/app/pages/authorizer/context/store'
import { AuthorizerTableStoreProvider } from 'v2/app/pages/authorizer/context'
import { DepositStore } from 'v2/app/pages/accounts/pages/banks/context/store'
import { DepositStoreProvider } from 'v2/app/pages/accounts/pages/banks/context'
import { ServicesProvider } from 'v2/services/useServices'
import { renderHook, RenderHookResult } from '@testing-library/react-hooks'
import { BreadcrumbsProvider } from 'v2/hooks/useBreadcrumbs'

const generateClassName = createGenerateClassName({
  productionPrefix: 'ix'
})

const BaseProviders: React.FC = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={Themes.default}>
        <BreadcrumbsProvider>
          <Router history={history}>{children}</Router>
        </BreadcrumbsProvider>
      </ThemeProvider>
    </StylesProvider>
  )
}

const customRenderer = (
  ui: any,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => {
  return render(ui, { wrapper: BaseProviders, ...options })
}

export const testComponentId = 'test-component'

export const fakeUserStore: Partial<UserStore> = {
  activeTab: 0,
  setActiveTab: jest.fn()
}

export const renderWithUserStore = (
  ui: any,
  store?: Partial<UserStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <UserProvider value={{ ...fakeUserStore, ...store }}>
        {children}
      </UserProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakePasswordResetStore: Partial<PasswordResetStore> = {
  currentStep: PasswordResetStep.Request,
  setCurrentStep: jest.fn()
}

export const renderWithPasswordResetStore = (
  ui: any,
  store?: Partial<PasswordResetStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <PasswordResetProvider value={{ ...fakePasswordResetStore, ...store }}>
        {children}
      </PasswordResetProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeAuthorizerTableStore: Partial<AuthorizerTableStore> = {
  idKey: '__',
  uri: '/api/source',
  _getItemId: jest.fn()
}

export const renderWithAuthorizerTableStore = (
  ui: any,
  store?: Partial<AuthorizerTableStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <AuthorizerTableStoreProvider
        value={{ ...fakeAuthorizerTableStore, ...store }}
      >
        {children}
      </AuthorizerTableStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeDepositStore: Partial<DepositStore> = {}

export const renderWithDepositStore = (
  ui: any,
  store?: Partial<DepositStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <DepositStoreProvider value={{ ...fakeDepositStore, ...store }}>
        {children}
      </DepositStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const renderHookWithServiceProvider = (
  hookFn: any,
  store: object = {}
): RenderHookResult<any, any> => {
  const WithServiceProvider: React.FC = ({ children }) => (
    <ServicesProvider value={store}>{children}</ServicesProvider>
  )

  return renderHook(hookFn, { wrapper: WithServiceProvider })
}

export * from '@testing-library/react'
export { customRenderer as render }
