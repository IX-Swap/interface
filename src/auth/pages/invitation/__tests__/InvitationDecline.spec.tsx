import React from 'react'
import { render } from 'test-utils'
import { InvitationDeclie } from 'auth/pages/invitation/InvitationDecline'

describe('InvitationDecline', () => {
  it('should match snapshot', () => {
    const { container } = render(<InvitationDeclie />)
    expect(container).toMatchSnapshot()
  })
})
