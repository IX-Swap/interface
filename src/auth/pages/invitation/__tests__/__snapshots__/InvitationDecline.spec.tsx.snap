// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvitationDecline should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-wrapper-1 css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <img
        alt="logo"
        src="image-mock"
      />
      <div
        class="MuiBox-root css-12aw9rk"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter makeStyles-description-3 css-1gd5lqg-MuiTypography-root"
      >
        Are you sure you want to decline the invitation?
      </p>
      <div
        class="MuiBox-root css-12aw9rk"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-99mzem-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-1jpm9pd"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root makeStyles-customButton-6 css-18vbx94-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Yes, Decline
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
